pkg/api/scylla/v1/cluster_types.go

```diff
@@ -95,6 +95,44 @@ type ScyllaClusterSpec struct {
 	// used for pulling Scylla and Agent images.
 	// +optional
 	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+
+	// ingressOptions is an optional Ingress configuration options.
+	// If provided, Ingress objects are generated for each Scylla node with following options.
+	// Experimental - subject to incompatible changes.
+	// +optional
+	IngressOptions *IngressOptions `json:"ingressOptions,omitempty"`
+}
+
+// IngressOptions defines configuration options for Ingress objects associated with cluster nodes.
+// Experimental - subject to incompatible changes.
+type IngressOptions struct {
+	// domain specifies DNS suffix of hostname of each Scylla node.
+	// It is concatenated with node host_id to form Ingress rule pointing to particular Scylla node.
+	Domain string `json:"domain"`
+
+	// ServiceBackendPort a port name or port number of the member service.
+	ServiceBackendPort ServiceBackendPort `json:"serviceBackendPort"`
+
+	// ingressClassName specifies Ingress class name.
+	// +optional
+	IngressClassName string `json:"ingressClassName,omitempty"`
+
+	// annotations specify custom annotations added to every Ingress object.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// ServiceBackendPort is the service port being referenced.
+type ServiceBackendPort struct {
+	// Name is the name of the port on the Service.
+	// This is a mutually exclusive setting with "Number".
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// Number is the numerical port number on the Service.
```

Comment:
Imo 'numerical' implies numerical types, not just `int32`. I'd change it to integer/integral or remove the adjective altogether, the type documents it well enough.